lizaku/assignment2/Q2.py

```diff
@@ -0,0 +1,31 @@
+from Q1 import Node, BinaryTree, create_tree
+
+
+def find_lca(cur_node, node1, node2):
+    # This algorithm is designed as follows: I start with the root and 
+    # try to check whether one node is present in the left subtree and other node
+    # is present in the right subtree. If this is the case, then the current node
+    # is LCA. If the traversal reaches the leaves and the value is not found, the recursion step
+    # returns None. If required values are not found in the right subtree,
+    # Then LCA should be found in the left subtree, and I start to inspect it.
+    # Otherwise, I start to inspect the right subtree.
+    if cur_node is None:
+        return None
```

Comment:
Can you raise a KeyError when at least one of the values is not found in the tree?